core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveIntegralCodec.java

```diff
@@ -248,6 +266,136 @@ public double decodeDouble(float value) {
     public double decodeDouble(double value) {
       throw new RuntimeException("internal error: " + debugInfo());
     }
+
+    @Override
+    public void decodeAndFillVector(ColumnPage columnPage, ColumnVectorInfo vectorInfo) {
+      CarbonColumnVector vector = vectorInfo.vector;
+      BitSet nullBits = columnPage.getNullBits();
+      DataType dataType = vector.getType();
+      DataType type = columnPage.getDataType();
+      int pageSize = columnPage.getPageSize();
+      BitSet deletedRows = vectorInfo.deletedRows;
+      fillVector(columnPage, vector, dataType, type, pageSize, vectorInfo);
+      if (deletedRows == null || deletedRows.isEmpty()) {
+        for (int i = nullBits.nextSetBit(0); i >= 0; i = nullBits.nextSetBit(i + 1)) {
+          vector.putNull(i);
+        }
+      }
+    }
+
+    private void fillVector(ColumnPage columnPage, CarbonColumnVector vector, DataType dataType,
```

Comment:
For Timestamp type 
`vector.putLong(i, byteData[i] * 1000);`  should be changed to `vector.putLong(i, (long) byteData[i] * 1000L);` otherwise it would cross integer range and give wrong results

Reply:
ok